### ISSUE TITLE
[4.x]: billingSameAsShipping setting is not persisting

### ISSUE THREAD

**Issue by Nellyaa** [craftcms/commerce#2866]
### Description

If you set 'billingAddressSameAsShipping' to true on the cart you get the expected response (billingSameAsShipping is true and the shipping address will be copied to the billing address). But if you reload the cart afterwards the setting will be 'false' again. So we have no way of knowing if the customer set their billing address to 'same as shipping' or his actual saved address.

### Steps to reproduce

1. Create a shipping address on a cart
2. Set billingAddressSameAsShipping to true and send source id of address to be copied
3. See billingSameAsShipping is true and address is copied
4. Reload the cart
5. See billingSameAsShipping is false again

### Expected behavior

The setting billingSameAsShipping stays true as long as no further changes to the addresses are being made (e.g. another address chosen, or the address is being edited)

### Actual behavior

The setting billingSameAsShipping becomes false after the next reload.

### Craft CMS version

4.0.5

### Craft Commerce version

4.0.3

### Installed plugins and versions

-

**Comment by pixelmachine** [comment]
@Nellyaa it not very intuitive but this is actually as intended, the billingAddressSameAsShipping value is not saved to the order. If you search here you'll find it discussed at length.

You can check if the addresses are the same by comparing them e.g.:

```php
cart.shippingAddress.sameAs(cart.billingAddress) or 
cart.billingAddressId == cart.shippingAddressId
```

I suppose you could also add a custom field to the order (a light switch field perhaps) and set that if you want to store the user preference.

**Comment by pixelmachine** [comment]
I suppose because once the order is completed there's really no reason to ever need to refer back to that value, you have the addresses saved so it doesn't really matter.

I ended up with some fairly awkward logic around if/which addresses are selected but I suppose you could save the value to a cookie in case the user comes and goes from that point in the checkout and you want to maintain their selection.

**Comment by Nellyaa** [comment]
Thank you @nfourtythree, this sounds exactly like what we need! :)

**Comment by lukeholder** [comment]
This is now fixed in 4.1 see changelog here: https://github.com/craftcms/commerce/blob/develop/CHANGELOG.md#410---2022-07-19
